include/flyft/data_layout.h

```diff
@@ -3,24 +3,39 @@
 
 #include <iterator>
 #include <type_traits>
+#include <vector>
 
 namespace flyft
     {
+//! DataLayout
+/*!
+ * Generates data structure for the custom data types used in the software.
+ * The class provides get and set functions for the array index and implements N-dimensional array
+ * mapping into one dimensional array using strided indexing scheme.
+ */
```

Comment:
The short description should be what the object is (the class name will be part of what doxygen generates), then the long description should describe what the class does (potentially with examples longer term)
```suggestion
//! Multidimensional array layout
/*!
 * DataLayout maps an N-dimensional index to one-dimensional
 * index using row-major ordering. Negative indexes are supported
 * and interpreted as relative to the zero-index.
 */
```